src/web/teap/teap/nextcloud/api.py

```diff
@@ -0,0 +1,52 @@
+from flask import Blueprint, jsonify, request, abort
+from flask.views import MethodView
+
+from nextcloud import NextCloud
+
+blueprint = Blueprint('nextcloud_api', __name__, url_prefix='/api')
+
+
+class NextCloudMixin:
+
+    @property
+    def nextcloud(self):
+        """ Get nextcloud instance """
+        # tmp mock nextcloud credentials
+        url = 'http://localhost:8080'
+        username = 'admin'
+        password = 'admin'
+        if url is None or username is None or password is None:
+            return abort(400)
+        nxc = NextCloud(endpoint=url, user=username, password=password)
+        return nxc
+
+
+class UserViewSet(NextCloudMixin,
```

Comment:
Just a clarification - TEAP won't need a REST API, it will use REST APIs of other services, while being a user-friendly "command center". Disregard this comment if you intend to feed output of these methods to client-side JS to reformat them.

Reply:
Yes, it is used on our vue.js front-end already